src/visitable.cpp

```diff
@@ -774,7 +782,8 @@ static int amount_of_internal( const T& self, const itype_id& id, bool pseudo, i
 {
     int qty = 0;
     self.visit_items( [&qty, &id, &pseudo, &limit] ( const item *e ) {
-        qty += ( e->typeId() == id && e->allow_crafting_component() && ( pseudo || !e->has_flag( "PSEUDO" ) ) );
+        bool is_component = e->typeId() == id && e->allow_crafting_component() && ( pseudo || !e->has_flag( "PSEUDO" ) );
+        qty = accumulate_with_limit( qty, is_component, limit );
```

Comment:
I still think the cast is clear given the context, but will bow to peer pressure.